### PR TITLE
Add Windows environment

### PR DESCRIPTION
**Pull request by Luni-4**
Add Windows environment in preparation for #7 

Unfortunately, some tests failed, but I don't know why. Any idea?

Thanks for your review!

### CODE REVIEW

**Comment by shssoichiro** [comment]
Looks like the tests are failing to parse some of the test y4m files on Windows. Off the top of my head I'm not sure why, but I have a VM I can use to investigate, unless you'd be willing to.

**Comment by Luni-4** [comment]
> Looks like the tests are failing to parse some of the test y4m files on Windows. Off the top of my head I'm not sure why, but I have a VM I can use to investigate, unless you'd be willing to.

I have a Windows machine, so I can investigate on it. What I found strange is that it happens for certain y4m files only, but I haven't studied the situation in depth yet.

**Comment by Luni-4** [comment]
The problem is upstream. I compared two y4m files, both of them produced by the `resize` example contained in the `y4m` repository. The first file has been generated on Windows, while the second one on Linux. They are different in size and content. 

This happens only with files whose chroma components have been subsampled though.

**Comment by shssoichiro** [comment]
This is very strange and somewhat concerning. There doesn't seem to be any windows- or unix-specific code in the y4m crate. So that means this may be a bug in the Rust std library? That seems really unlikely...

**Comment by Luni-4** [comment]
> This is very strange and somewhat concerning. There doesn't seem to be any windows- or unix-specific code in the y4m crate. So that means this may be a bug in the Rust std library? That seems really unlikely...

That's strange in fact, but neither do I think there's a bug in the Rust std library .

So, I bisect the `y4m` crate and I discovered that up till this [commit](https://github.com/image-rs/y4m/commit/56876fefcb43c23492621e64c2c4da481c496009), the output is correct on Windows. I mean, the file size is not 32 btyes, but 30 kb xD

Could you double-check with your VM, please? In this way, we are sure that I'm not doing anything wrong. Thanks!

**Comment by shssoichiro** [comment]
Turns out it was not a bug in the app, y4m, or Rust. Git, it turns out, does not recognize y4m files, and treats them as text files. This means that when you check them out on Windows, Git thinks it should change the line endings. It turns out we need to tell Git that y4m files are binary, so I've added a gitattributes file to do this.

39mw5r

File: .github/workflows/ci-build.yml [hunk 1]
```diff
@@ -3,14 +3,17 @@ name: ci
 on:
   pull_request:
     branches:
-      - master
+      - windows-actions
```
Comment (lu-zero) [inline .github/workflows/ci-build.yml]: Before-merge this should be changed back.

Comment (Luni-4) [reply .github/workflows/ci-build.yml]: Done, thanks!

**Comment by Luni-4** [comment]
> Turns out it was not a bug in the app, y4m, or Rust. Git, it turns out, does not recognize y4m files, and treats them as text files. This means that when you check them out on Windows, Git thinks it should change the line endings. It turns out we need to tell Git that y4m files are binary, so I've added a gitattributes file to do this.

Incredible! So it was Git the great enemy of this PR xD 

Thanks for your fix and support @shssoichiro

**Pull request merged**